src/ripple/rpc/tests/AccountOffers.test.cpp

```diff
@@ -0,0 +1,275 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2016 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/protocol/JsonFields.h>     // jss:: definitions
+#include <ripple/protocol/Feature.h>
+#include <ripple/test/jtx.h>
+
+namespace ripple {
+namespace test {
+
+class AccountOffers_test : public beast::unit_test::suite
+{
+public:
+    static
+    std::unique_ptr<Config>
+    makeConfig(bool setup_admin)
+    {
+        auto p = std::make_unique<Config>();
+        setupConfigForUnitTests(*p);
+        // the default config has admin active
+        // ...we remove them if setup_admin is false
+        if (! setup_admin)
+        {
+            (*p)["port_rpc"].set("admin","");
+            (*p)["port_ws"].set("admin","");
+        }
+        return p;
+    }
+
+    // test helper
+    bool check_array_size(const Json::Value & val, unsigned int size)
+    {
+        return expect(val.isArray()) &&
+               expect(val.size() == size);
+    }
+
+    // test helper
+    bool check_marker(const Json::Value& val, bool is_present)
```

Comment:
I am slightly allergic to functions which accept a boolean argument and have one of two behaviors. Thoughts on renaming this to `expect_marker (Json const& val)` and calling it when a marker is expected? When it isn't, we just fall back to `expect (! val.isMember (jss::marker));`.